### ISSUE TITLE
Review Authentication and authorization concept doc

### ISSUE THREAD

**Issue by shilpa-padgaonkar** [camaraproject/WorkingGroups#27]
https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/Working/CAMARA-AuthN-AuthZ-Concept.md

**Comment by shilpa-padgaonkar** [comment]
Comments by @jordonezlucena from Issue #9 have been moved here as Issue #9 will be closed with the deliverable being contributed and comments regarding AuthN-AuthZ concept will be tracked in this new review issue.
It is assumed that there is a need to have an 'exposure function / API gateway' mediating between API consumer (aka API invoker, within 3rd party domain) and API producer (within operator domain). This function is in charge of implementing necessary access control (AuthN/Z), and providing service API registry and discovery capabilities.

Some requirements in relation to AuthN/Z here:

Mutual authentication between the API consumer and the 'exposure function / API GW ' should be supported.
The transport messages exchanged between the API consumer and the 'exposure function / API GW' should be integrity and confidentiality protected.
The API provider shall determine whether API consumer is authorised to access service API.
NOTE 1: API GW have baseline/basic built-in auth capabilities. There are other external solutions (ForgeRock, Auth0) that can be integrated into the API GW to provide API provider with enhanced security management functionality.
Possible solutions for Auth/Z:

Using TLS-PSK -> assumes that both the API consumer and API producer are pre-provisioned with certificates.
Using PKI
Using TLS with OAuth 2.0 token -> preferred solution.
All of these solutions are covered with CAPIF framework, which is proposed as reference solution for 'exposure function / API GW) in CAMARA. For further details, see issue https://github.com/camaraproject/WorkingGroups/issues/1

**Comment by jordonezlucena** [comment]
The link at the beginning is now broken -- needs to be updated

**Comment by T-sm** [comment]
> The link at the beginning is now broken -- needs to be updated
Link updated, the document has been moved to [Working](https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/Working) directory in line with issue #35 request.

**Comment by jordonezlucena** [comment]
Master document!! Congratulations for the result -- comprehensive tutorial-style document that help make sure we're all on the same page. 

I agree with the @Kevsy suggestions. A couple of additional suggestions from my side:

1) In the "CAPIF" clause, it'd be good to reference the document that surveys the usage of CAPIF in CAMARA. The document, available at [https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/Working/API-exposure-reference-solution.docx], is linked to issue #1.  I can take action on this, if you like. 

2) There are well-defined concepts defined in the document (e.g., resource owner, resource server, access token, authentication, authorization, authorization server, etc...). Does it make sense to move these concepts to the standard resource description doc in issue #10?

**Comment by T-sm** [comment]
Thank you for the review and your inputs.
Closing this issue as agreed during May 4th Commonalities Workgroup meeting.
A UEId addendum will be prepared within a separate issue.

**Comment by jpengar** [comment]
Hi, update proposal included in PR https://github.com/camaraproject/WorkingGroups/pull/76 
(thanks @jordonezlucena for helping us with PR creation, we still don't have permission as contributor :S)

They are actually some minor changes.

### Details about the change:

- Proposed to remove sentence "The Camara Service APIs in most use cases will make use of the client crendentials grant.". We do believe that this could be potencially not 100% true according to legal teams (every time some personal data is processed even when it is provided like an API input e.g. IP address it would require the user consent legally speaking). So it's more than enough what is perfectly explained in doc by:

```
Client Credentials grant: (when user resource is not involved & the client belongs to confidential category)
Authorization Code grant: (when user resource is involved)
```

- User consent Management proposal is treated as a separated document (https://github.com/camaraproject/WorkingGroups/pull/68) but we have added OIDC CIBA flow and jwt-bearer as part of the Relevant authorization grants and flows. It is part of the User Consent Management proposal as well.

- Added clarification about user identity in authentication and the concept of "end user" in the context of CAMRA APIs being exposed by Telco operators. Being different from UE identification included in PR https://github.com/camaraproject/WorkingGroups/pull/51

**Comment by jordonezlucena** [comment]
This issue gets closed again, once PR #80 has been merged
